### ISSUE TITLE
Develop Tag Plugin System: Enhance Subjects page by Integrating Tags

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#8279]
Once the basic ability to CRUD (create, read, update, delete) Tags has been achieved in #7766 and basic infrastructure is in place to connect/associate subject strings and Tags, the next step is to create a generic schema / DSL (domain specific language) for Tags that lets librarians define metadata to enhance how the subject page renders.

@JaydenTeoh has drafted/prototyped a plugins idea in #7928 that lets librarians associate Tags with any number of plugins (which are ostensibly macros or templates). #7928 is an amazing proof of concept, which achieves the goal of extending the capabilities of subject pages by defining a list of plugins which the subject page integrates. The concept is close but I think we may want to abstract as much of the implementation/code logic **out** of the Tag document and into the subject page (as a means of defining an implementation of the relationships specified within the Tag).

At minimum, we want to be able to (a) add more named carousels to Subject pages and (b) present patrons with a few librarian-chosen facets (`related_tags` or `related_subjects`) -- similar to the search page --  that the patron can click on to narrow their search.

This issue has 3 parts, all of which are prototyped by #7928:
* [ ] Prototype Tag Schema MVP -- schema/DSL that allows librarians to add queries and relationships to a Tag. Each Tag might have...
  * A `primary_query` for the Tag/subject (by default / if not specified is just `subject:whatever-the-subject-is`)
  * interesting `related_tags` a patron might click to drill down and further narrow what is shown by the `primary_query`
* [ ] Update the create/edit Tag UI that enables librarians to specify Tag plugins to customize a subject page'
* [ ] Integrate the subject page to implement each Tag plugin   

![image](https://github.com/internetarchive/openlibrary/assets/978325/25a2b955-c70f-4933-a14f-8c757d019c09)

![image](https://github.com/internetarchive/openlibrary/assets/978325/5ce3d79e-d1a9-4855-a053-e7e7cb25d4e4)

## Notes
<img width="572" alt="Screenshot 2023-09-18 at 3 28 45 PM" src="https://github.com/internetarchive/openlibrary/assets/978325/7e09c674-7a03-4e4e-92db-be4ea3285054">

**Comment by mekarpeles** [comment]
#8466

**Comment by mekarpeles** [comment]
I think once we land #8466 we'll be in great shape to discuss next steps.

**Comment by jimchamp** [comment]
https://github.com/internetarchive/openlibrary/labels/State%3A%20Blocked on approach.  It's unclear how these plugins will be used when rendering these new subject pages.  Will the page be composed entirely of the `plugins` that are on the `Tag`?  Or, are the plugins meant to be rendered in a specific section of the subject page?

For now, subject `Tag`s will include a `body` field, which will contain an HTML string for rendering the new subject page.  A separate issue will be created to track this work.

**Comment by mekarpeles** [comment]
@Chisomnwa related to our work together :)

**Comment by Chisomnwa** [comment]
Hi, @mekarpeles. This is nice! This definitely builds on the tags work I'm doing! I've seen the plugin prototype in #7928 and this makes sense as the next step after the genre backfill. Happy to help break this down when the time comes.
